src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -22,16 +27,37 @@
      */
     public static final int DEFAULT_MAX_STRING_LEN = 1_000_000;
 
+    protected final int _maxDepth;
     protected final int _maxNumLen;
     protected final int _maxStringLen;
 
     private static final StreamReadConstraints DEFAULT =
-        new StreamReadConstraints(DEFAULT_MAX_NUM_LEN, DEFAULT_MAX_STRING_LEN);
+        new StreamReadConstraints(DEFAULT_MAX_DEPTH, DEFAULT_MAX_NUM_LEN, DEFAULT_MAX_STRING_LEN);
 
     public static final class Builder {
+        private int maxDepth;
         private int maxNumLen;
         private int maxStringLen;
 
+        /**
+         * Sets the maximum depth. The depth is a count of objects and arrays that have not been closed,
+         * `{` and `[` respectively.
+         *
+         * @param maxDepth the maximum depth
+         *
+         * @return this builder
+         * @throws IllegalArgumentException if the maxDepth is set to a negative value
+         *
+         * @since 2.15
+         */
+        public Builder maxDepth(final int maxDepth) {
```

Comment:
`maxDepth` could work; another possibility `maxNesting`. Not sure what terminology is most commonly used with JSON parsers.

Reply:
I've changed this to maxNestingDepth